v2/bundle/x509bundle/bundle.go

```diff
@@ -1,65 +1,182 @@
 package x509bundle
 
 import (
+	"bytes"
 	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
 	"io"
+	"io/ioutil"
+	"reflect"
+	"sync"
 
 	"github.com/spiffe/go-spiffe/v2/spiffeid"
 )
 
+const certType string = "CERTIFICATE"
+
 // Bundle is a collection of trusted public key material for a trust domain.
 type Bundle struct {
+	tdMtx       *sync.RWMutex
+	trustDomain spiffeid.TrustDomain
+
+	rootsMtx *sync.RWMutex
```

Comment:
> nit: the prevailing guidance is that mutex fields in structs that are used via references (i.e. *Bundle) be values, not pointers.

Thanks! will fix

> I'm trying to image a scenario where a bundle consumer is also mutating the bundle...

Hmm... I've been thinking about his... One case that comes to mind is someone serving a bundle endpoint (not backed by the workload API), that runs a separate thread to update the roots certificates in the bundle source. Does it makes sense?

Reply:
Yeah, I think you are right. Considering this can be used as a bundle source and provided to a TLS config that will be hitting this thing from another goroutine, it's probably better to be safe.